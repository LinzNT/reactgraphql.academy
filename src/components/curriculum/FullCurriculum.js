import React from 'react'
import { Col, Row } from '../layout/Grid'
import { H2Ref, H4 } from '../text'
import Link from '../navigation/Link'
import { Tabs, TabList, TabItem, TabContent, ContentItem } from '../layout/Tabs'
import {
  CurriculumReactBootcamp,
  CurriculumPartTime,
  CurriculumAdvancedReact,
  CurriculumGraphQLBootcamp,
  CurriculumReactFundamentals,
  CurriculumGraphQLWorkshops,
  CurriculumGraphQLAPI,
} from './index'

import CurriculumReactWorkshops from './CurriculumReactWorkshops'
import {
  REACT_NATIVE,
  REACT_BOOTCAMP,
  REACT_FUNDAMENTALS,
  PART_TIME,
  ADVANCED_REACT,
  GRAPHQL_BOOTCAMP,
  GRAPHQL_WORKSHOP,
  GRAPHQL_API,
} from '../../config/data'

class FullCurriculum extends React.Component {
  state = {
    active: REACT_BOOTCAMP,
  }

  setActive = active => {
    this.setState({ active })
  }

  render() {
    const { trainings } = this.props
    const commonCurriculumProps = {
      trainings,
      showTitle: false,
    }
    return (
      <React.Fragment>
        <Row>
          <Col lg={10} lgOffset={1}>
            <H2Ref>
              Choose a course...{' '}
              <Link to="#curriculum" name="curriculum">
                #
              </Link>
            </H2Ref>
          </Col>
        </Row>
        <Row>
          <Col lg={11}>
            <Tabs onChange={this.setActive} active={this.state.active}>
              <TabList offset>
                <TabItem name={REACT_BOOTCAMP}>React Bootcamp</TabItem>
                <TabItem name={REACT_FUNDAMENTALS}>React Fundamentals</TabItem>
                <TabItem name={ADVANCED_REACT}>Advanced React</TabItem>
                <TabItem name={PART_TIME}>Part-time Course</TabItem>
                <TabItem name={REACT_NATIVE}>React Workshops</TabItem>
              </TabList>
              <TabList offset>
                <TabItem name={GRAPHQL_BOOTCAMP}>GraphQL Bootcamp</TabItem>
                <TabItem name={GRAPHQL_API}>GraphQL API</TabItem>

                <TabItem name={GRAPHQL_WORKSHOP}>GraphQL Workshops</TabItem>
              </TabList>

              <TabContent>
                <ContentItem name={REACT_BOOTCAMP}>
                  <CurriculumReactBootcamp {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={REACT_FUNDAMENTALS}>
                  <CurriculumReactFundamentals {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={ADVANCED_REACT}>
                  <CurriculumAdvancedReact {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={PART_TIME}>
                  <CurriculumPartTime {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={REACT_NATIVE}>
                  <CurriculumReactWorkshops {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={GRAPHQL_BOOTCAMP}>
                  <CurriculumGraphQLBootcamp {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={GRAPHQL_API}>
                  <CurriculumGraphQLAPI {...commonCurriculumProps} />
                </ContentItem>
                <ContentItem name={GRAPHQL_WORKSHOP}>
                  <CurriculumGraphQLWorkshops {...commonCurriculumProps} />
                </ContentItem>
              </TabContent>
            </Tabs>
          </Col>
        </Row>
      </React.Fragment>
    )
  }
}

export default FullCurriculum
