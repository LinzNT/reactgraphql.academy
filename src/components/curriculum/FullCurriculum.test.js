import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import FullCurriculum from './FullCurriculum'

describe('<FullCurriculum />', () => {
  it('should render component', () => {
    const wrapper = shallow(<FullCurriculum />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
