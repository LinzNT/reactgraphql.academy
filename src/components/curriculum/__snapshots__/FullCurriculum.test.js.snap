// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullCurriculum /> should render component 1`] = `
<Fragment>
  <Row>
    <Styled(Col)
      lg={10}
      lgOffset={1}
      md={12}
      xs={12}
    >
      <Styled(styled.h1)>
        Our React curriculum
         
        <Link
          name="curriculum"
          to="#curriculum"
        >
          #
        </Link>
      </Styled(styled.h1)>
    </Styled(Col)>
  </Row>
  <CurriculumBootcamp
    showTitle={false}
  />
</Fragment>
`;
